Homework6/ConsoleApp1/Classes/Logger.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Text;
+using System.Reflection;
+using System.Text.Json;
+using Attributes;
+
+namespace Task1.Classes
+{
+ public class Logger
+ {
+  public string JSONName { get; private set; }
+
+  public Logger(string name) => JSONName = name;
+
+  public void Track<T>(params T[] objects)
+  {
+   List<PropertyInfo> GetTrackingProperties<T>()
+   {
+    List<PropertyInfo> list = new();
+    PropertyInfo[] propertyInfos = typeof(T).GetProperties();
+
+    foreach (var prop in propertyInfos)
+    {
+     object[] attributes = prop.GetCustomAttributes(true);
+     foreach (var attr in attributes)
+     {
+      if (attr is TrackingPropertyAttribute)
+      {
+       list.Add(prop);
+      }
+     }
+    }
+
+    return list;
+   }
+
+   List<FieldInfo> GetTrackingFields<T>()
+   {
+    List<FieldInfo> list = new();
+    FieldInfo[] fieldInfos = typeof(T).GetFields();
+
+    foreach (var field in fieldInfos)
+    {
+     object[] attributes = field.GetCustomAttributes(true);
+     foreach (var attr in attributes)
+     {
+      if (attr is TrackingPropertyAttribute)
+      {
+       list.Add(field);
+      }
+     }
+    }
+
+    return list;
+   }
+
+   string GetSerializedObjectStringByPropertyList<T>(List<PropertyInfo> customPropAttributes, List<FieldInfo> customFieldAttributes, params T[] objects)
```

Comment:
rename parameter customPropAttributes to propertiesWithAttributes. 
The name is confusing, this is not a list of attributes, this is a list of  properties

Reply:
rename parameter customFieldAttributes  to fieldsWithAttributes. 
The name is confusing, this is not a list of attributes, this is a list of  fields